fundamentals/middleware/rate-limit/WebRateLimitAuth/Program.cs

```diff
@@ -0,0 +1,447 @@
+app.Configuration.GetSection(MyRateLimitOptions.MyRateLimit).Bind(myOptions);
+
+app.UseRateLimiter(new RateLimiterOptions()
+    .AddConcurrencyLimiter(policyName: getPolicyName,
+          new ConcurrencyLimiterOptions(permitLimit: myOptions.permitLimit,
+          queueProcessingOrder: QueueProcessingOrder.OldestFirst,
+          queueLimit: myOptions.queueLimit))
+    .AddPolicy(policyName: postPolicyName, partitioner: httpContext =>
+    {
+        string userName = httpContext?.User?.Identity?.Name ?? string.Empty;
+
+        if (!StringValues.IsNullOrEmpty(userName))
+        {
+            return RateLimitPartition.CreateTokenBucketLimiter(userName, key =>
+                new TokenBucketRateLimiterOptions(tokenLimit: myOptions.tokenLimit2,
+                    queueProcessingOrder: QueueProcessingOrder.OldestFirst,
+                    queueLimit: myOptions.queueLimit,
+                    replenishmentPeriod: TimeSpan.FromSeconds(myOptions.replenishmentPeriod),
+                    tokensPerPeriod: myOptions.tokensPerPeriod,
+                    autoReplenishment: myOptions.autoReplenishment));
+        }
+        else
+        {
+            return RateLimitPartition.CreateTokenBucketLimiter("Anon", key =>
+                new TokenBucketRateLimiterOptions(tokenLimit: myOptions.tokenLimit,
+                    queueProcessingOrder: QueueProcessingOrder.OldestFirst,
+                    queueLimit: myOptions.queueLimit,
+                    replenishmentPeriod: TimeSpan.FromSeconds(myOptions.replenishmentPeriod),
+                    tokensPerPeriod: myOptions.tokensPerPeriod,
+                    autoReplenishment: true));
+        }
+    }));
+// </snippet_adm2>
+
+static string GetUserEndPointMethod(HttpContext context) =>
+    $"Hello {context.User?.Identity?.Name ?? "Anonymous"} " +
+    $"Endpoint:{context.Request.Path} Method: {context.Request.Method}";
+
+app.MapGet("/test", (HttpContext context) => $"{GetUserEndPointMethod(context)}")
+                                        .RequireRateLimiting(getPolicyName);
+
+app.MapRazorPages().RequireRateLimiting(getPolicyName)
+                   .RequireRateLimiting(postPolicyName);
+
+app.MapDefaultControllerRoute();
+
+app.Run();
+// </snippet_adm>
+#elif JWT
+// <snippet_jwt>
+using System.Threading.RateLimiting;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.RateLimiting;
+using Microsoft.Extensions.Primitives;
+using WebRateLimitAuth.Models;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthorization();
+builder.Services.AddAuthentication("Bearer").AddJwtBearer();
+
+var app = builder.Build();
+
+app.UseAuthorization();
+
+var jwtPolicyName = "jwt";
+var postPolicyName = "post";
+var myOptions = new MyRateLimitOptions();
+app.Configuration.GetSection(MyRateLimitOptions.MyRateLimit).Bind(myOptions);
+
+app.UseRateLimiter(new RateLimiterOptions()
+    .AddNoLimiter(policyName: jwtPolicyName)
```

Comment:
What's the intention here? Adding a no limiter is functionally equivalent to not adding any endpoint limiter for this endpoint - the global limiter will still run. We're removing `AddNoLimiter` in rc1 (we are also adding an attribute that allows you to disable rate limiting entirely for an endpoint, including the global limiter)